tests/draft2019-09/optional/format/time.json

```diff
@@ -5,9 +5,74 @@
         "tests": [
             {
                 "description": "a valid time string",
+                "data": "08:30:06Z",
+                "valid": true
+            },
+            {
+                "description": "a valid time string with leap second",
+                "data": "23:59:60Z",
+                "valid": true
+            },
+            {
+                "description": "a valid time string with second fraction",
                 "data": "08:30:06.283185Z",
                 "valid": true
             },
+            {
+                "description": "a valid time string with plus offset",
+                "data": "08:30:06+00:20",
+                "valid": true
+            },
+            {
+                "description": "a valid time string with minus offset",
+                "data": "08:30:06-08:00",
+                "valid": true
+            },
+            {
+                "description": "a valid time string with case-insensitive Z",
+                "data": "08:30:06z",
+                "valid": true
+            },
+            {
+                "description": "an invalid time string with invalid hour",
+                "data": "24:00:00Z",
+                "valid": false
+            },
+            {
+                "description": "an invalid time string with invalid minute",
+                "data": "00:60:00Z",
+                "valid": false
+            },
+            {
+                "description": "an invalid time string with invalid second",
+                "data": "00:00:61Z",
+                "valid": false
+            },
+            {
+                "description": "an invalid time string with invalid leap second (wrong hour)",
+                "data": "22:59:60Z",
```

Comment:
A variant of this test could be added, to be extra-tricky, that uses a leap hour at midnight in the local timezone (but is invalid, because they can only be added at midnight UTC).

Reply:
I can add `15:59:60-08:00` (which is extracted from one of examples in [rfc3330 section 5.8](https://tools.ietf.org/html/rfc3339#section-5.8) as a case in a different time offset.

Reply:
As for local timezone, I guess you're thinking of something using the [unknown local offset](https://tools.ietf.org/html/rfc3339#section-4.3) `-00:00`:

> If the time in UTC is known, but the offset to local time is unknown, this can be represented with an offset of "-00:00".  This differs semantically from an offset of "Z" or "+00:00", which imply that UTC is the preferred reference point for the specified time.  RFC2822 [IMAIL-UPDATE] describes a similar convention for email.

I'm not sure we're able to validate leap seconds here because the validation might be run with a different timezone than the user inputting the time. So sounds like we should allow leapseconds at any time with that offset because the "unknown offset" could be anything. Does that sound reasonable?

Reply:
No, not quite.

I'll write up some tests of these cases for a seperate PR. What you have here so far is fine.

Reply:
Add leap second test case with time offset in 7bde0bf. Will wait and see what you have regarding local time.